Rewrite help page section. (#2821)
pallets/click: docs/documentation.rst

Help Pages
===================

.. currentmodule:: click

Click makes it very easy to document your command line tools. For most things Click automatically generates help pages for you. By design the text is customizable, but the layout is not.

Help Texts
----------

Commands and options accept help arguments. For commands, the docstring of the function is automatically used if provided.

Simple example:

.. click:example::

    @click.command()
    @click.option('--count', default=1, help='number of greetings')
    @click.option('--name', help='a name')
    def hello(count, name=False):
        """This script prints hello and a name one or more times."""
        for x in range(count):
            if name:
                click.echo(f"Hello {name}!")
            else:
                click.echo("Hello!")

> And from the command line:

Author: Great idea.

        that will be rewrapped again.
        """

And from the command line:

.. click:run::

    invoke(cli, args=['--help'])

To change the rendering maximum width, pass ``max_content_width`` when calling the command.

.. code-block:: python

    cli(max_content_width=120)

.. _doc-meta-variables:

Truncating Help Texts
---------------------

Click gets :class:`Command` help text from the docstring. If you do not want to include part of the docstring, add the ``\f`` escape marker to have Click truncate the help text after the marker.

Example:

.. click:example::

    @click.command()
    def cli():
        """First paragraph.
        \f

        Words to not be included.
        """

And from the command line:

.. click:run::

    invoke(cli, args=['--help'])

Placeholder Variable
-----------------------

> The default placeholder variable (`meta variable <https://en.wikipedia.org/wiki/Metasyntactic_variable#IETF_Requests_for_Comments>`_) in the help pages is the parameter name in uppercase with underscores. This can be changed for Commands and Parameters with the ``options_metavar`` and  ``metavar`` kwargs.

Author: Going to leave this alone.

    invoke(hello, args=['--help'])

Help Parameter Customization
----------------------------

> The help parameter(s) is automatically added by Click for any command. The default is ``--help`` but can be override by the context setting :attr:`~Context.help_option_names`. Click also performs automatic conflict resolution on the default help parameter so if a command itself implements a parameter named ``help`` then the default help will no be run.

Reviewer: ```suggestion
Help parameters are automatically added by Click for any command. The default is ``--help`` but can be override by the context setting :attr:`~Context.help_option_names`. Click also performs automatic conflict resolution on the default help parameter so if a command itself implements a parameter named ``help`` then the default help will not be run.
```
